docs/user/ppl/admin/datasources.rst

```diff
@@ -25,44 +25,91 @@ Definitions of datasource and connector
+* Datasource Creation POST API ("_plugins/_query/_datasources") ::
 
+    POST https://localhost:9200/_plugins/_query/_datasources
+    content-type: application/json
+    Authorization: Basic {{username}} {{password}}
 
-[**To be run on all the nodes in the cluster**] Command to add datasources.json file to OpenSearch Keystore ::
+    {
+        "name" : "my_prometheus",
+        "connector": "prometheus",
+        "properties" : {
+            "prometheus.uri" : "http://localhost:8080",
+            "prometheus.auth.type" : "basicauth",
+            "prometheus.auth.username" : "admin",
+            "prometheus.auth.password" : "admin"
+        },
+        "allowedRoles" : ["prometheus_access"]
+    }
+
+* Datasource modification PUT API ("_plugins/_query/_datasources") ::
+
+    PUT https://localhost:9200/_plugins/_query/_datasources
+    content-type: application/json
+    Authorization: Basic {{username}} {{password}}
+
+    {
+        "name" : "my_prometheus",
+        "connector": "prometheus",
+        "properties" : {
+            "prometheus.uri" : "http://localhost:8080",
+            "prometheus.auth.type" : "basicauth",
+            "prometheus.auth.username" : "admin",
+            "prometheus.auth.password" : "admin"
+        },
+        "allowedRoles" : ["prometheus_access"]
+    }
+
+* Datasource Read GET API("_plugins/_query/_datasources/{{dataSourceName}}" ::
+
+    GET https://localhost:9200/_plugins/_query/_datasources/my_prometheus
+    content-type: application/json
+    Authorization: Basic {{username}} {{password}}
 
-    >> bin/opensearch-keystore add-file plugins.query.federation.datasource.config datasources.json
+  **Authentication Information won't be vended out in GET API's response.**
 
-Datasources can be configured during opensearch start up or can be updated while the opensearch is running.
-If we update a datasource configuration during runtime, the following api should be triggered to update the query engine with the latest changes.
+* Datasource Deletion DELETE API("_plugins/_query/_datasources/{{dataSourceName}}") ::
 
-[**Required only if we update keystore settings during runtime**] Secure Settings refresh api::
+    DELETE https://localhost:9200/_plugins/_query/_datasources/my_prometheus
+    content-type: application/json
+    Authorization: Basic {{username}} {{password}}
 
-    >> curl --request POST \
-      --url http://{{opensearch-domain}}:9200/_nodes/reload_secure_settings \
-      --data '{"secure_settings_password":"{{keystore-password}}"}'
+Authorization of datasource configuration APIs
+==============================================
+Each of the datasource configuration management apis are controlled by following actions respectively.
 
+* cluster:admin/opensearch/datasources/create [Create POST API]
+* cluster:admin/opensearch/datasources/read   [Get GET API]
+* cluster:admin/opensearch/datasources/update [Update PUT API]
+* cluster:admin/opensearch/datasources/delete [Delete DELETE API]
```

Comment:
do we need add following actions in OSD security plugin configuration?

Reply:
yes..I have changes ready. I need to add and will add ppl as well.